src/pqm/pqm.py

```diff
@@ -86,56 +193,81 @@ def _pqm_test(
         warnings.warn(
             "Number of samples is small (less than twice the number of reference samples). Result will have high variance and/or be non-discriminating."
         )
-    if whiten:
+    if z_score_norm:
         mean, std = _mean_std(x_samples, y_samples)
         y_samples = (y_samples - mean) / std
         x_samples = (x_samples - mean) / std
 
     # Determine fraction of x_samples to use as reference samples
     x_frac = nx / (nx + ny) if x_frac is None else x_frac
 
-    # Determine number of samples from each distribution (x_samples, y_samples, gaussian)
-    Nx, Ny, Ng = np.random.multinomial(
-        num_refs,
-        [x_frac * (1.0 - gauss_frac), (1.0 - x_frac) * (1.0 - gauss_frac), gauss_frac],
+    # Determine number of samples from each distribution
+    probs = torch.tensor(
+        [
+            x_frac * (1.0 - gauss_frac),
+            (1.0 - x_frac) * (1.0 - gauss_frac),
+            gauss_frac,
+        ],
+        device=device,
+    )
+
+    counts = Multinomial(total_count=num_refs, probs=probs).sample()
```

Comment:
I think the numpy version is easier to use in one line here, rather than using pytorch then converting tolist in the end